examples/md-flexible/src/TimeDiscretization.cpp

```diff
@@ -30,7 +41,29 @@ void calculatePositionsAndResetForces(autopas::AutoPas<ParticleType> &autoPasCon
     v *= deltaT;
     f *= (deltaT * deltaT / (2 * m));
     const auto displacement = v + f;
+#ifdef AUTOPAS_ENABLE_DYNAMIC_CONTAINERS
     iter->setR(iter->getR() + displacement);
+#else
+    // Sanity check that particles are not too fast for the Verlet skin technique. Only makes sense if skin > 0.
+    if (not iter->addRDistanceCheck(displacement, maxAllowedDistanceMovedSquared) and
+        maxAllowedDistanceMovedSquared > 0) {
+      const auto distanceMoved = std::sqrt(dot(displacement, displacement));
+      // If this condition is violated once this is not necessarily an error. Only if the total distance traveled over
+      // the whole rebuild frequency is farther than the skin we lose interactions.
+      AUTOPAS_OPENMP(critical)
+      std::cerr << "A particle moved farther than verletSkinPerTimestep/2: " << distanceMoved << " > "
+                << autoPasContainer.getVerletSkin() / autoPasContainer.getVerletRebuildFrequency()
+                << "/2 = " << maxAllowedDistanceMoved << "\n"
+                << *iter << "\nNew Position: " << iter->getR() + displacement << std::endl;
+      if (fastParticlesThrow) {
+        throwException = true;
+      }
+    }
+  }
+
+  if (throwException) {
+    throw std::runtime_error("At least one particle was too fast!");
+#endif
   }
```

Comment:
I don't really like that this `}` is for different things depending on the CMake variable. If `AUTOPAS_ENABLE_DYNAMIC_CONTAINERS`, then this is for the above `if (throwException)`; otherwise, it is for the `for` loop. I would personally keep the same `}` to end the `for` loop and then have a separate CMake branch around this `if` statement, even if it does mean extra CMake variable branching.

Reply:
done, see [4da4355](https://github.com/AutoPas/AutoPas/pull/821/commits/4da43550d82b376771ada1352a61fc648ff9e0d9)